packages/react-components/react-table/src/hooks/types.ts

```diff
@@ -12,8 +13,14 @@ export interface SortState {
 export interface ColumnDefinition<TItem> {
   columnId: ColumnId;
   compare?: (a: TItem, b: TItem) => number;
+  renderHeader?: () => React.ReactNode;
+  renderCell?: (item: TItem) => React.ReactNode;
 }
 
+export interface ColumnDefinitionRender<TItem>
+  extends Omit<ColumnDefinition<TItem>, 'renderHeader' | 'renderCell'>,
+    Required<Pick<ColumnDefinition<TItem>, 'renderHeader' | 'renderCell'>> {}
```

Comment:
This is so hard to read sometimes I wonder if it's not better to simply repeat the properties on a new type like:

```ts
export interface ColumnDefinitionRender<TItem> {
  columnId: ColumnId;
  compare?: (a: TItem, b: TItem) => number;
  renderHeader: () => React.ReactNode;
  renderCell: (item: TItem) => React.ReactNode;
}
```